V3-Upgrade-Guide.md

```diff
@@ -221,11 +221,47 @@ The `serviceMapping` option in OData generator has been renamed to `optionsPerSe
 The `service-mapping.json` file is now `options-per-service.json`.
 The file will not be generated by default in the input directory, as in the previous versions.
 
-Set the `optionsPerService` to either:
+You **must** set  the `optionsPerService` to have the generator consider the options. 
+You can set the value to either:
 
 - The file path containing the options per service (e.g. `options.json`).
-  If the file does not exist, it will be created.
-- The directory from which the file is read/created (e.g. `someDir`. This will create a file named `options-per-service.json` in `someDir`)
+  If the file does not exist, it will be created and initialized.
+  If the file exists, missing or partial service options will be added with the default values. 
+- The directory from which the file is read/created (e.g. `someDir`. This will read/create a file named `options-per-service.json` in `someDir`)
+
+Also, the properties in the configuration have changed:
+- The option `serviceName` is removed.
+A value for the documentation header is derived from the directory name.
+- The `npmPackageName` is renamed to `packageName` to align with the OpenApi generator.
+- The keys change to the relative paths of the service specifications.
+In case you have problems finding the relative path, rerun the generator which will add new default entries to the exising options.
+
+Here is an example how the options change:
+
+```json
+//Old Format
+{
+  "API-TEST_SRV": {
+    "directoryName": "test-service",
+    "basePath": "/odata/test-service",
+    "npmPackageName": "npm-package-test-service",
+    "serviceName": "some-text-used-in-documentation-service"
+  }
+}
+
+//New Format
+{
+  "../test-resources/odata-spec/API-TEST_SRV.edmx": {
```

Comment:
this path is relative to what? the package.json?

Reply:
we use `process.cwd()` which means relative to the location the generator is executed from. We could also make it relative to the location of the options-per-serivice so the file location where the path is given. I have no strong opinion on it.

Reply:
thinking about it I guess making it relative to options per service might be more stable.. unless one moves that file..

I'm not sure. We can also leave it like it is..